plasma_framework/contracts/src/exits/payment/controllers/PaymentProcessInFlightExit.sol

```diff
@@ -283,13 +286,17 @@ library PaymentProcessInFlightExit {
 
             // If the input has been challenged, isInputPiggybacked() will return false
             if (token == withdrawal.token && exit.isInputPiggybacked(i)) {
-                bool successBondReturn = SafeEthTransfer.transferReturnResult(
-                    withdrawal.exitTarget, withdrawal.piggybackBondSize, self.safeGasStipend
-                );
 
-                // we do not want to block a queue if bond return is unsuccessful
-                if (!successBondReturn) {
-                    emit InFlightBondReturnFailed(withdrawal.exitTarget, withdrawal.piggybackBondSize);
+                if (withdrawal.piggybackBondSize > withdrawal.bountySize) {
```

Comment:
Not really understanding this. When will this happen?

Oh is this is for safe math revert protection and not that we have idea when this will happen?

Reply:
This is to not attempt returning when the `bondReturnValue` is 0. 
If `piggybackBondSize == bountySize` then don't attempt returning the bond and give away the bounty
Could have also used `withdrawal.piggybackBondSize != withdrawal.bountySize` but since bond >= bounty this could be more clear?
I should add a comment here.

Reply:
oh.....probably we should consider to reject the possibility of `bond == bounty` for simplification? What is the tradeoff?

Reply:
Hmm.. I guess only in a situation when somehow the cost of processing exits is higher than what we want for the bond. (and would want to increase the bond size to cover the higher bounty) The bond already has a buffer in the calculations which the bounty doesn't so it is very unlikely that we would need this, but still a possibility?

Reply:
However, we could still have the strict condition of bond > bounty for a scenario where the above happens. The only downside i guess is if the size of bond and bounty is close and the difference is low the exit initiator gets a very low bond value back.

Reply:
added a comment here to follow while updating